File: atcoder/python/abc210_219/abc212_B.py

import sys
infile = sys.stdin.buffer

def gs()  : return infile.readline().rstrip()
def gi()  : return int(gs())
def gf()  : return float(gs())
def gss() : return gs().split()
def gis() : return [int(x) for x in gss()]
def gfs() : return [float(x) for x in gss()]

def main(infn="") :
    global infile
    infile = open(infn,"r") if infn else open(sys.argv[1],"r") if len(sys.argv) > 1 else sys.stdin
    X = gs()
    if X[0] == X[1] == X[2] == X[3] :
        ans = "Weak"
    else :
        weak = True
        XX = [int(c) for c in X]
        for i in range(3) :
            if (XX[i] + 1) % 10 != XX[i+1] : weak = False
        ans = "Weak" if weak else "Strong"
    sys.stdout.write(str(ans)+'\n')

if __name__ == '__main__' :
    main()
    sys.stdout.flush()

File: abc/go/abc212/abc212_B/abc212_B.go
package main

import (
	"bufio"
	"fmt"
	"os"
)
var wrtr = bufio.NewWriterSize(os.Stdout, 10_000_000)
var rdr = bufio.NewScanner(os.Stdin)
func gs() string  { rdr.Scan(); return rdr.Text() }
func main() {
	//f1, _ := os.Create("cpu.prof"); pprof.StartCPUProfile(f1); defer pprof.StopCPUProfile()
	defer wrtr.Flush()
	infn := ""; if infn == "" && len(os.Args) > 1 {	infn = os.Args[1] }
	if infn != "" {	f, e := os.Open(infn); if e != nil { panic(e) }; rdr = bufio.NewScanner(f) }
	rdr.Split(bufio.ScanWords); rdr.Buffer(make([]byte,1024),1_000_000_000)
	// PROGRAM STARTS HERE
	S := gs(); ans := "Strong"
	SS := []int{0,0,0,0}; for i:=0;i<4;i++ { SS[i] = int(S[i]-'0') }
	if SS[0] == SS[1] && SS[0] == SS[2] && SS[0] == SS[3] { ans = "Weak" }
	if SS[1] == (SS[0]+1)%10 && SS[2] == (SS[1]+1)%10 && SS[3] == (SS[2]+1)%10 { ans = "Weak" }
	fmt.Println(ans)
}

